.github/workflows/main.yaml

```diff
@@ -91,7 +91,20 @@ jobs:
 
           echo "waiting 5m for test to pass..."
           kubectl wait al --for=condition=Deployed --all --timeout=5m || (kubectl get po -A && false)
+      - name: Orphan Integration Test
+        run: |
+          echo "setting up test..."
+          kubectl apply -f ./testdata/addons-orphan/addons-source.yaml
+          kubectl patch gitrepository addons-config -n gotk-system -p "{\"spec\": {\"ref\": {\"branch\": \"$GITHUB_HEAD_REF\"}}}" --type=merge
+          kubectl apply -f ./testdata/addons-orphan/addons-repo.yaml
+          kubectl apply -f ./testdata/addons-orphan/addons.yaml
 
+          echo "waiting 5m for test to pass..."
+          kubectl wait al --for=condition=Deployed --all --timeout=5m || (kubectl get po -A && false)
+          sleep 1
+          kubectl get al
+          kubectl get helmreleases.helm.toolkit.fluxcd.io -A  -o jsonpath='{range .items[*]}{.metadata.name}{"\t"}{.metadata.namespace}{"\t"}{.metadata.labels}{"\t"}{.metadata.ownerReferences[0].name}{"\t"}{"\n"}{end}'
+          kubectl get addonslayers.kraan.io -A  -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{range .status.resources[*]}{.name}{"\t"}{.namespace}{"\n"}{end}{"\n"}{end}'
```

Comment:
This prints the output but requires a beautiful human being to verify its correctness. I suggest adding a grep or some conditional test that will pass/fail. Unless waiting for ALs to be deployed is enough validation

Reply:
Waiting for deployment should be enough, I added this to prove to myself it works, will remove it or do as you suggest and parse output

Reply:
removed